File: uncategorized/442_find_all_duplicates_in_an_array.ts
function findDuplicates(nums: number[]): number[] {

    let ans = [];
    let N = nums.length;

    for(let i=0; i<N; i++){
        let curr = nums[i];
        if(curr < 0){
            curr *= -1;
        };
        curr -= 1;
        if(nums[curr] < 0){
            ans.push(curr + 1);
        } else {
            nums[curr] = nums[curr] * -1;
        }
    };

    return ans;

};

File: uncategorized/442_find_all_duplicates_in_an_array.rs
impl Solution {
    pub fn find_duplicates(nums: Vec<i32>) -> Vec<i32> {

        let mut nums = nums;
        let mut ans:Vec<i32> = Vec::new();

        for i in 0.. nums.len() {
            let mut curr = nums[i];
            if curr < 0 {
                curr *= -1;
            }
            curr -= 1;
            if nums[curr as usize] < 0 {
                ans.push(curr + 1);
            } else {
                nums[curr as usize] *= -1;
            }
        };

        ans

    }
}